everviz/plugins/objectives_plot/objectives_plot.py

```diff
@@ -71,38 +86,77 @@ def layout(self):
 
     def set_callbacks(self, app):
         @app.callback(
-            Output(self.graph_id, "figure"), [Input(self.radio_id, "value"),],
+            Output(self.graph_id, "figure"),
+            [Input(self.function_dropdown_id, "value"), Input(self.radio_id, "value"),],
         )
-        def update_graph(graph_type):
-            df = get_data(self.data_path)
+        def update_graph(func_list, radio_value):
+            # # The key_list arguments is the list of functions to plot.
+            if func_list is None:
+                return {}
+
+            # # Get the data, setting its index.
+            if radio_value == "Statistics":
+                data = get_data(self.statistics_file).set_index("function")
+            else:
+                data = get_data(self.values_file).set_index("function")
+
+            # Make a cycle iterator over the plotly colors.
+            colors = cycle(DEFAULT_PLOTLY_COLORS)
+
             traces = []
-            if graph_type == "Statistics":
-                for label in ["P10", "Mean", "P90"]:
-                    fill = "tonexty" if label in ["P90", "Mean"] else None
+            for color, key in zip(colors, func_list):
+                # Select all data belonging to the current key.
+                func_data = data.xs(key)
+
+                # Make the traces, with mean, P10 and P90, shading in between.
+                if radio_value == "Statistics":
+                    traces.extend(
+                        [
+                            go.Scatter(
+                                y=func_data["P90"],
+                                x=func_data["batch"],
+                                mode="lines",
+                                marker={"size": 10},
+                                line={"color": color},
+                                name=key + "(P90)",
+                                showlegend=False,
+                            ),
+                            go.Scatter(
+                                y=func_data["P10"],
+                                x=func_data["batch"],
+                                mode="lines",
+                                line={"color": color},
+                                marker={"size": 10},
+                                fill="tonexty",
+                                name=key + "(P10)",
+                                showlegend=False,
+                            ),
+                            go.Scatter(
+                                y=func_data["Mean"],
+                                x=func_data["batch"],
+                                mode="lines",
+                                marker={"size": 10},
+                                line={"color": color},
+                                name=key,
+                                showlegend=True,
+                            ),
+                        ]
+                    )
+                else:
```

Comment:
The two plugins are now looking quite similar and do the same, they might not be able to be completely merged, but could the parts that overlap be split into shared functions?

Reply:
That is worth a try, but last time I tried this, it was quite a task. So, I would rather leave that for a separate issue/pull request.